### ISSUE TITLE
Using nullable types - and no access to pull requests

### ISSUE THREAD

**Issue by joachimboggild** [craigbridges/Nettle#2]
Hi, I get an exception in the Common/Conversion/GenericObjectToTypeConverter.cs class. It happens when I make a custom function where I need to handle nullable types. In my case, I have made a FormatDecimalFunction, which should format a decimal value according to a format string. The value, however, might be null, so I have specified the value as nullable, like this: 

```
using System;
using System.Globalization;
using Nettle;
using Nettle.Compiler;
using Nettle.Functions;

namespace Global.CustomerNotifier.HtmlServices.Builders.Soa.NettleFunctions
{
	public sealed class FormatDecimalFunction : FunctionBase
	{
		public FormatDecimalFunction() : base()
		{
			DefineRequiredParameter
			(
				"Number",
				"The decimal number to format",
				typeof(decimal?)
			);
			DefineRequiredParameter
			(
				"Format",
				"The format to apply",
				typeof(string)
			);
		}
		protected override object GenerateOutput(TemplateContext context, params object[] parameterValues)
		{
			Validate.IsNotNull(context);

			var number = GetParameterValue<decimal?>
			(
				"Number",
				parameterValues
			);
			var format = GetParameterValue<string>
			(
				"Format",
				parameterValues
			);

			return number?.ToString(format, new CultureInfo("en-GB")) ?? "";
		}

		public override string Description => "Formats a decimal value.";
	}
}

```

This function produces the exception when being used in a template.

It can be fixed by using the technique presented here: https://stackoverflow.com/questions/18015425/invalid-cast-from-system-int32-to-system-nullable1system-int32-mscorlib

I have changed the code to this effect, but I am not sure how I can make a pull request towards your repository. Do you need to enable pull requests for this to work?

Sincerely,
Joachim Bøggild

**Comment by craigbridges** [comment]
Hi Joachim,

Thanks for finding (and fixing) this bug. I didn't realise I needed to enable pull requests for you to create one, I thought they were enabled by default. I think I have now enabled pull requests Can you try again?

Thanks,
Craig

**Comment by joachimboggild** [comment]
Yes, I did manage to make a pull request now. Thanks!

**Comment by joachimboggild** [comment]
Hello again,

I just tried to switch from the local version of Nettle (i.e. cloned from GitHub) to NuGet. But it seems that this change is not in NuGet yet. Would you by any chance have plans to publish a new NuGet version soon? :-)

**Comment by craigbridges** [comment]
Ah yes sorry, I had completely forgotten to publish it to NuGet - I think because NuGet wanted me to create a Microsoft account at the time, so I left it and then forget!

I have now published version 1.2.4 to NuGet.

**Comment by joachimboggild** [comment]
Excellent, thanks very much!
